torch/csrc/jit/passes/quantization.cpp

```diff
@@ -421,14 +428,31 @@ Node* insertQuantDeQuantCall(
 
   Node* quant = g->create(at::Symbol::aten(quantize_func), inputs);
   quant->output()->setDebugName(v->debugName() + ".quant");
-
-  Node* dequant = g->create(at::Symbol::aten("dequantize"), {quant->output()});
-  dequant->output()->setDebugName(v->debugName() + ".dequant");
-
   g->insertNode(quant);
-  g->insertNode(dequant);
 
-  return dequant;
+  // Insert all the dequants first
+  Node* observer_node = nullptr;
+  while (!v->uses().empty()) {
```

Comment:
This loop is pretty convoluted, why not just loop over all uses and for each use add a dequantize node? maybe we can discuss in person tomorrow

Reply:
We have no such API for replacing the usage. We only have replaceAll and replaceFirst.

Reply:
does `replaceInput`/`replaceInputWith` help? https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/ir.h#L445

Reply:
This will probably become simpler now after PR30140 landed. There should be no need to look through uses.

Reply:
we discussed offline. we'll have two loops, first loop will store all the nodes that uses the value and second loop with add quant/dequant and replace the value with use `replaceInputWith`